java/arcs/core/storage/CapabilitiesResolver.kt

```diff
@@ -13,25 +13,64 @@ package arcs.core.storage
 
 import arcs.core.common.ArcId
 import arcs.core.data.Capabilities
+import arcs.core.data.Schema
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
 import arcs.core.util.TaggedLog
 
 /**
  * [CapabilitiesResolver] is a factory class that creates [StorageKey]s based on the registered
  * providers and given [Capabilities].
  */
 class CapabilitiesResolver(
-    val options: StorageKeyOptions,
+    val options: CapabilitiesResolverOptions,
     private val creators: StorageKeyCreatorsMap = getAllCreators()
 ) {
     private val log = TaggedLog { "CapabilitiesResolver" }
 
     /* Options used to construct [CapabilitiesResolver] */
-    data class StorageKeyOptions(val arcId: ArcId)
+    data class CapabilitiesResolverOptions(val arcId: ArcId)
+
+    /**
+     * Options passed to registered [StorageKey] constructors.
+     * @property arcId An identifier of an Arc requesting the [StorageKey]
+     * @property entitySchema A schema of an entities that will be stored
+     * @property unique A unique component of the [StorageKey]
+     * @property location A memory location of the [StorageKey]
+     */
+    interface StorageKeyOptions {
+        val arcId: ArcId
+        val entitySchema: Schema
+        val unique: String
+            get() = TODO("not implemented")
+        val location: String
+            get() = TODO("not implemented")
```

Comment:
Rather than default implementations throwing `NotImplementedError` or whatever `TODO` throws... should these be nullable/empty?

Reply:
done, thanks